Готовая работа к уроку №2

GromZDev opened the pull request:
Сделал ДЗ. Реализовал переход на новый экран через router.navigateTo. Для этого добавил в интерфейс и его реализацию новую функцию fun userDetails(bundle: Bundle): Screen, куда в Презентере и будем передавать сущность через Parcelable. А в новом фрагменте UserDetailsFragment мы её подцепляем и сетим.

GromZDev committed:
Переделал проект под списки пользователей гита. Сделал свой кастомный itemView. Подкрутил к проекту Чичерон.

GromZDev committed:
Реализовал переход на новый экран через router.navigateTo. Для этого добавил в интерфейс и его реализацию новую функцию fun userDetails(bundle: Bundle): Screen, куда в Презентере и будем передавать сущность через Parcelable. А в новом фрагменте UserDetailsFragment мы её подцепляем и сетим.

GromZDev committed:
Причесал код, удалил уже ненужный класс FragmentMain и его лэйаут. Поставил полям доступы private.

PeterBukhal-Geekbrains commented on app/src/main/java/q3_kotlin/popular_libraries/myapplication/view/MainActivity.kt:
```diff
@@ -0,0 +1,64 @@
+
+        vb = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(vb?.root)
+//
```
никогда не стоит оставлять в коде закомментированный код, никогда!! если необходимо сохранить что то лучше использовать git stash ли что то похожее.

PeterBukhal-Geekbrains commented on app/src/main/java/q3_kotlin/popular_libraries/myapplication/presenter/UsersPresenter.kt:
```diff
@@ -0,0 +1,53 @@
+
+            //TODO: переход на экран пользователя c помощью router.navigateTo
+            val currentUser = usersListPresenter.users[itemView.pos]
+            val bundle = Bundle()
```
Напоминаю что мы не используем Android зависимости в коде презентеров. Это грубое нарушение MVP.

PeterBukhal-Geekbrains commented on app/src/main/java/q3_kotlin/popular_libraries/myapplication/presenter/UsersPresenter.kt:
```diff
@@ -0,0 +1,53 @@
+class UsersPresenter(private val usersRepo: GithubUsersRepo, private val router: Router) :
+    MvpPresenter<UsersView>() {
+
+    val screens = AndroidScreens() // Для перехода во фрагмент с описанием пользователя
```
Следует передавать в конструкторе, в MainPresenter он как раз у тебя передается, почему бы не сделать и тут, только в виде интерфейса конечно.

PeterBukhal-Geekbrains commented on app/src/main/java/q3_kotlin/popular_libraries/myapplication/view/UserDetailsFragment.kt:
```diff
@@ -0,0 +1,47 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val user = arguments?.getParcelable<GithubUser>(BUNDLE_EXTRA)
```
представление не должно ничего запрашивать у презентера, но только презентер сообщает view, что делать, то что сделано у тебя это прямое нарушение MVP. Нет презентера вовсе в данном случае.

PeterBukhal-Geekbrains reviewed (changes requested):
Есть критические замечания. Пока не принимаю.

GromZDev committed:
Переделал проект согласно рекомендациям:
1) Сделал свой презентер для окна с описанием юзера UserDetailsPresenter;
2) Также организовал отдельный View к нему - UserView. В конструктор передаем нашу модель;
3) Убрал ненужный закомментированный код;
4) Показываем теперь логин пользователя через функцию интерфейса UserView.

PeterBukhal-sovcombank reviewed (approved):
Отлично, задание выполнено, замечаний нет.
